### PR TITLE
Examples for using Storj core

### LINKED ISSUE #316: Add Library Usage Documentation
We have complete documentation on the low level protocol, and plenty of documentation on the API. What we don't have is much documentation on how it ties together into something functional. We should add the following tutorials with complete scripts that the user can run:

- [ ] Authentication to the bridge
- [ ] Encrypting the data
- [ ] Uploading the data
- [ ] Download the data 
- [ ] Decrypting the data 

Might want to make a generic template for this so we can do this for all languages. Will be working with @Miskerest to do the same for the Python library that is currently being revamped.

### PR DESCRIPTION
**Pull request by aleitner**
* Closes #316

### CODE REVIEW

**Comment by super3** [comment]
@aleitner This is good from my side. @bookchin Review and merge plz.

**Comment by bookchin** [comment]
File need to be renamed to use kebab-case due to inconsistencies on case insensitive file systems.

File: example/3-authenticate-with-keypair.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+var fs = require('fs');
+// Set the bridge api URL
+var api = 'https://api.storj.io';
+
+// Load keypair from your saved private key
+var keypair = storj.KeyPair(fs.readFileSync('./private.key').toString());
+
+// Login using the keypair generated
+var client;
+client = storj.BridgeClient(api, {keypair: keypair});
```
Comment (bookchin) [inline example/3-authenticate-with-keypair.js]: Why assign to undefined then assign to bridge client?

File: example/4b-add-remove-keys.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  // Remove Public Key that was just added
+  client.destroyPublicKey(keypair.getPublicKey(), function(err) {
+    if (err) {
+      // Handle error on failure.
+      return console.log('error', err.message);
+    }
+
+    console.log('info', 'Key successfully revoked.');
+  });
+
```
Comment (bookchin) [inline example/4b-add-remove-keys.js]: Extraneous line break

File: example/6a-upload-file.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            'info',
+            'Name: %s, Type: %s, Size: %s bytes, ID: %s',
+            [file.filename, file.mimetype, file.size, file.id]
+          );
+
+          // Mirror the item
+
+        }
+      );
+      }
```
Comment (bookchin) [inline example/6a-upload-file.js]: Some weird spacing, closing stuff here.

File: example/6b-download-file.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+var target = fs.createWriteStream(filepath);
+
+//
+var secret = keyring.get(id);
+
+// Prepare to decrypt the encrypted file
+var decrypter = new storj.DecryptStream(secret);
+var received = 0;
+// list of servers to exclude when finding the download server
+var exclude = '';
```
Comment (bookchin) [inline example/6b-download-file.js]: This should be an `[]`

File: example/6c-list-bucket-files.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  // Log out info for each file
+  files.forEach(function(file) {
+    console.log(
+      'info',
+      'Name: %s, Type: %s, Size: %s bytes, ID: %s',
+      [file.filename, file.mimetype, file.size, file.id]
+    );
+  });
+
```
Comment (bookchin) [inline example/6c-list-bucket-files.js]: Extraneous line break

**Pull request merged**